integration/test/Test/Conversation.hs

```diff
@@ -410,19 +411,33 @@ testAddUnreachable = do
 testAddingUserNonFullyConnectedFederation :: HasCallStack => App ()
 testAddingUserNonFullyConnectedFederation = do
   let overrides =
-        def {dbBrig = setField "optSettings.setFederationStrategy" "allowAll"}
-          <> fullSearchWithAll
-  startDynamicBackends [overrides] $ \domains -> do
-    own <- make OwnDomain & asString
-    other <- make OtherDomain & asString
-    [alice, alex, bob, charlie] <-
-      createAndConnectUsers $ [own, own, other] <> domains
-
-    let newConv = defProteus {qualifiedUsers = [alex]}
+        def
+          { dbBrig =
+              setField "optSettings.setFederationStrategy" "allowDynamic"
+                >=> removeField "optSettings.setFederationDomainConfigs"
+          }
+  startDynamicBackends [overrides] $ \[dynBackend] -> do
+    own <- asString OwnDomain
+    other <- asString OtherDomain
+
+    -- Ensure that dynamic backend only federates with own domain, but not other
+    -- domain.
+    --
+    -- FUTUREWORK: deleteAllFedConns at the time of acquiring a backend, so
+    -- tests don't affect each other.
+    deleteAllFedConns dynBackend
+    void $ createFedConn dynBackend (FedConn own "full_search")
```

Comment:
Can you please explain how this dance of deleting and then immediately making a connection works?

Reply:
If you have two tests: one of them connects `dynBackend1` with all backends and second one only connects `dynBackend`1 with `OwnDomain`. if the first test runs first, the second test will always fail. So, the second test must always delete all the connections `dynBackend1` has and then only create a connection with `OwnDomain` to ensure that it is not affected by the first test.

Reply:
I see.

Wasn't the plan to provide more isolation between test executions? This dance within the test is very fragile, it incurs what I hope can be called a needless brain load, and is not a general solution.

Is there something we can do in the (dynamic) backend setup that would make them isolated from each other (even if we go with e.g. randomly assigning identifiers (names) and such, which should be good enough if they are very unlikely to collide)?

Reply:
Yes, that's why I added the FUTUREWORK. While claiming an environment we could simply truncate the table which keeps this information around.